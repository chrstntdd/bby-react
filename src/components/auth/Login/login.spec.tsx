import * as React from 'react';
import { shallow, mount } from 'enzyme';
import * as configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { MemoryRouter as Router } from 'react-router-dom';
import createRouterContext from 'react-router-test-context';
import renderer from 'react-test-renderer';

import ConnectedLogin, { Login } from './';

import { Provider } from 'react-redux';

const setup = () => {
  const mockStore = configureStore();
  const initialState = {
    auth: {
      userProfile: null,
      jwt: null,
      error: '',
      message: '',
      content: '',
       isAuthenticated: false,
      waiting: false
    }
  };
  const context = createRouterContext();

  const props = {
    handleSubmit: jest.fn(),
    valid: false,
    loginInputs: [
      {
        name: 'employeeNumber',
        type: 'text',
        label: 'Employee Number'
      },
      {
        name: 'password',
        type: 'password',
        label: 'Password'
      }
    ]
  };
  const store = mockStore(initialState);
  const enzymeWrapper = mount(
    <Router context={context}>
      <Provider store={store}>
        <ConnectedLogin {...props} store={store} />
      </Provider>
    </Router>
  );

  return {
    props,
    enzymeWrapper,
    initialState
  };
};

describe('DUMB <Login/>', () => {
  it('should render without crashing', () => {
    const props = {
      handleSubmit: jest.fn()
    };
    const wrapper = shallow(<Login {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('SMART <ConnectedLogin/>', () => {
  it('should render the connected component', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper).not.toBe(null);
  });

  it('should render two inputs', () => {
    const { enzymeWrapper, props } = setup();

    const inputs = enzymeWrapper.find('input');

    expect(inputs.length).toEqual(2);
  });

  it('should dispatch handleSubmit when the form is submitted', () => {
    const { enzymeWrapper, props } = setup();
    const form = enzymeWrapper.find('form');
    form.simulate('submit');

    expect(props.handleSubmit).toHaveBeenCalled();
  });
});
