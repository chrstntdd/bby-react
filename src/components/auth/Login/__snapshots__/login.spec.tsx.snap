// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DUMB <Login/> should render without crashing 1`] = `
<section
  id="login-wrapper"
>
  <LoadingIndicator
    message="Signing you in now. Please wait."
    waiting={false}
  />
  <div
    className="show"
    id="login-card"
  >
    <h1>
      Sign In
    </h1>
    <Form
      handleSubmitCb={[Function]}
    >
      <div
        className="input-wrapper"
      >
        <Input
          autoComplete="off"
          inputRef={
            Object {
              "current": null,
            }
          }
          label="Employee Number"
          name="employeeNumberInput"
          type="text"
          validateFn={[Function]}
          validationCb={[Function]}
        />
        <span
          className="focus-border"
        />
      </div>
      <div
        className="input-wrapper"
      >
        <Input
          autoComplete="off"
          inputRef={
            Object {
              "current": null,
            }
          }
          label="Password"
          name="passwordInput"
          type="password"
          validateFn={[Function]}
          validationCb={[Function]}
        />
        <span
          className="focus-border"
        />
      </div>
      <button
        disabled={true}
        type="submit"
      >
        Login
      </button>
    </Form>
    <Link
      replace={false}
      to="/forgot-password"
    >
      Forgot password?
    </Link>
    <Link
      replace={false}
      to="/sign-up"
    >
      Sign Up
    </Link>
    <p>
      Just testing? Use the temporary credentials below.
    </p>
  </div>
  <div
    className="test-credentials"
  >
    <p>
      <span>
        Employee Number
      </span>
      : a1
    </p>
    <p>
      <span>
        Password
      </span>
      : testtesttest
    </p>
  </div>
</section>
`;
